developer/src/kmcmplib/src/kmx_u16.cpp

```diff
@@ -259,7 +259,7 @@ KMX_WCHAR * u16tok(KMX_WCHAR* p, const KMX_WCHAR* delim, KMX_WCHAR** ctx) {
 	if (*q) {
 		*q = 0;
 		q++;
-		while (u16chr(delim, *q)) q++;
+		while (*q && u16chr(delim, *q)) q++;
```

Comment:
This is a separate bug: it needs to be tracked in a separate PR. I think this is important for regression management.

Two further opportunities here:
1. Audit other uses of `u16chr` for the same bug elsewhere in the compiler.
2. Add a unit test for `u16tok`, looking at existing usage for current patterns.

Reply:
I recognize that this is my fault for saying 'put both of these bugs into one issue'. Sorry!

Reply:
I will leave a single issue ([#11814](https://github.com/keymanapp/keyman/issues/11814)), but break out a new PR for the missing second parameter bug.

Reply:
I have reversed the fix, although other checks in the index section of `GetXStringImpl()` prevent the bug from manifesting.